### PR TITLE
ZENKO-3056 Fix unexpected error messages

### PR DESCRIPTION
**Pull request by nicolas2bert**
- List object: Decrease the number of keys returned in the response -> **reduce server response time**. 
- Infinite scroll: force to only load 1 page of items at a time.
- Wait for objects to be loaded before accessing the first object metadata (if it exists).
- Improve AWS API error messages that are rendered.

### CODE REVIEW

File: src/react/databrowser/objects/ObjectListTable.jsx [hunk 1]
```diff
@@ -35,11 +36,20 @@ type Props = {
 export default function ObjectListTable({ objects, bucketName, toggled, isVersioningType, prefixWithSlash }: Props){
     const dispatch = useDispatch();
     const nextMarker = useSelector((state: AppState) => state.s3.listObjectsResults.nextMarker);
-    const isItemLoaded = index => {
-        const shouldRefetch = nextMarker && index === objects.size - 1;
+    const loading = useSelector((state: AppState) => state.networkActivity.counter > 0);
```
Comment (JBWatenbergScality) [inline src/react/databrowser/objects/ObjectListTable.jsx:39]: in order to simplify a bit the application you can create a `useTypedSelector` like so : 
```js
export const useTypedSelector: <TSelected>(
  selector: (state: AppState) => TSelected,
  equalityFn?: (left: TSelected, right: TSelected) => boolean
) => TSelected = useSelector;
```

That you can then reuse everywhere without the necessity to type explicitly the state parameter

File: src/react/databrowser/objects/ObjectListTable.jsx [hunk 2]
```diff
@@ -35,11 +36,20 @@ type Props = {
...
+    const objectsLength = objects.size;
+    const isToggledFull = toggled.size > 0 && toggled.size === objectsLength;
+
+    const isItemLoaded = useCallback(index => {
```
Comment (JBWatenbergScality) [inline src/react/databrowser/objects/ObjectListTable.jsx:43]: index parameter could be typed

**Review by JBWatenbergScality — APPROVED** [review]

File: src/react/databrowser/objects/ObjectListTable.jsx (hunk 2 above)
Comment (nicolas2bert) [reply src/react/databrowser/objects/ObjectListTable.jsx:43]: `flowtype/require-parameter-type` has not been enforced by our linter yet.

File: src/react/databrowser/objects/ObjectListTable.jsx (hunk 1 above)
Comment (nicolas2bert) [reply src/react/databrowser/objects/ObjectListTable.jsx:39]: Yes, I have a plan to do that using the helper type: `TypedUseSelectorHook`.

**Review by terrydervaux — APPROVED** [review]
LGTM 👍 

This fix seems to solve the issue reported in that comment: https://github.com/scality/zenko-ui/pull/69#pullrequestreview-554870406.

The navigation in the objects browsing pages is more fluid.

**Comment by nicolas2bert** [comment]
@bert-e approve

**Comment by nicolas2bert** [comment]
bert-e reset

**Pull request merged**